fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/impl/transfer/context/InitialContentsContainerItemContext.java

```diff
@@ -37,6 +37,22 @@ protected ItemVariant getBlankVariant() {
 		protected long getCapacity(ItemVariant variant) {
 			return Long.MAX_VALUE;
 		}
+
+		@Override
+		public long insert(ItemVariant insertedVariant, long maxAmount, TransactionContext transaction) {
+			StoragePreconditions.notBlankNotNegative(insertedVariant, maxAmount);
+
+			// Pretend we can't insert anything to route every insertion through insertOverflow.
+			return 0;
+		}
+
+		@Override
+		public long extract(ItemVariant extractedVariant, long maxAmount, TransactionContext transaction) {
+			StoragePreconditions.notBlankNotNegative(extractedVariant, maxAmount);
+
+			// Pretend we can extract anything, but never actually do it.
+			return maxAmount;
+		}
```

Comment:
Hmm, isn't this a change to `ContainerItemContext.withInitial` behaviour as well?

Reply:
That's true, but we're still within its (somewhat vague) contract:

```java
 	/**
	 * Return a context that can accept anything, and will accept (and destroy) any overflow items, with some initial variant and amount.
	 * This can typically be used to check if a variant provides an API, or simulate operations on the returned API,
	 * for example to simulate how much fluid could be extracted from the variant and amount.
	 */
```

It depends whether you read `with some initial variant and amount` as "initially the context has this content, and whatever happens after that is undefined", or as "the context always has the initial content whatever happens". I think the latter makes more sense.

Reply:
To me "initial" implies that there can be other content that isn't "initial", otherwise it's pointless to have that word there 😛

IMO the new behaviour would be something like `withConstant`

Reply:
Hmmm, what would happen to `withInitial` after mutating the stack would still not be defined. How about deprecating `withInitial` in favor of `withConstant` ? (Both with the new behavior)

Reply:
Went for it, but keeping the old behavior for `withConstant` just in case.